File: src/main/java/com/github/kayjamlang/core/containers/ConstructorContainer.java
package com.github.kayjamlang.core.containers;

import com.github.kayjamlang.core.Stmt;
import com.github.kayjamlang.core.Type;
import com.github.kayjamlang.core.expressions.data.Argument;
import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.opcodes.AccessType;

import java.util.Collections;
import java.util.List;

/**
 * Constructor of class
 */
public class ConstructorContainer extends FunctionContainer {

    /**
     * Create constructor for class
     * @param arguments Arguments of constructor
     * @param children Code in constructor
     * @param line Line of constructor
     */
    public ConstructorContainer(List<Argument> arguments, List<Stmt> children, int line) {
        super(null, children, AccessType.NONE, arguments, Type.VOID, Collections.emptyList(), line);
    }
}

File: src/main/scala/com/github/kayjamlang/core/containers/ConstructorContainer.scala
package com.github.kayjamlang.core.containers

import com.github.kayjamlang.core.AdvancedMutableList
import com.github.kayjamlang.core.expressions.Expression
import com.github.kayjamlang.core.expressions.data.Argument
import com.github.kayjamlang.core.opcodes.AccessType

class ConstructorContainer(val arguments: AdvancedMutableList[Argument], children: AdvancedMutableList[Expression], accessType: AccessType, line: Int) extends Container(children, accessType, line)